torch/_logging/_internal.py

```diff
@@ -0,0 +1,424 @@
+
+    def get_log_qnames(self):
+        return set(self.log_alias_to_log_qname.values())
+
+    def get_artifact_log_qnames(self):
+        return set(self.artifact_log_qnames)
+
+    def get_child_log_qnames(self):
+        return set(self.child_log_qnames)
+
+    def is_off_by_default(self, artifact_qname):
+        return artifact_qname in self.off_by_default_artifact_names
+
+
+@dataclass
+class LogState:
+    # qualified log names -> currently set log level
+    log_qname_to_level: Dict[str, str] = field(default_factory=dict)
+
+    # the set of currently enabled artifacts
+    artifact_names: Set[str] = field(default_factory=set)
+
+    def enable_artifact(self, artifact_name):
+        self.artifact_names.add(artifact_name)
+
+    def is_artifact_enabled(self, name):
+        return name in self.artifact_names
+
+    def enable_log(self, log_qname, log_level):
+        self.log_qname_to_level[log_qname] = log_level
+
+    def get_log_level_pairs(self):
+        return self.log_qname_to_level.items()
+
+    def clear(self):
+        self.log_qname_to_level.clear()
+        self.artifact_names.clear()
+
+
+log_registry = LogRegistry()
+log_state = LogState()
+
+
+# User API for setting log properties
+# ex. format set_logs(LOG_NAME=LEVEL, ARTIFACT_NAME=bool)
+# ex. set_logs(dynamo=logging.DEBUG, graph_code=True)
+def set_logs(
+    dynamo=DEFAULT_LOG_LEVEL,
+    aot=DEFAULT_LOG_LEVEL,
+    inductor=DEFAULT_LOG_LEVEL,
+    bytecode=False,
+    aot_forward_graph=False,
+    aot_backward_graph=False,
+    aot_joint_graph=False,
+    graph=False,
+    graph_code=False,
+    guards=False,
+    output_code=False,
+    schedule=False,
+):
+    """
+    Enable setting the log level of individual components through kwargs.
+    Args are set using the following format:
+        set_logs(<log_name>=<log_level>,...<artifact_name>=<True or False>)
+    """
+    # ignore if env var is set
+    if LOG_ENV_VAR in os.environ:
+        return
+
+    log_state.clear()
+
+    def _set_logs(**kwargs):
+        for alias, val in kwargs.items():
+            if log_registry.is_artifact(alias):
+                if val:
+                    log_state.enable_artifact(alias)
+            elif log_registry.is_log(alias):
+                if val not in logging._levelToName:
+                    raise ValueError(
+                        f"Unrecognized log level for log {alias}: {val}, valid level values "
+                        f"are: {','.join([str(k) for k in logging._levelToName.keys()])}"
+                    )
+                if val != DEFAULT_LOG_LEVEL:
+                    log_state.enable_log(
+                        log_registry.log_alias_to_log_qname[alias], val
+                    )
+            else:
+                # Check if it is a qualified name log
+                # if so, check that its root logger is parent
+                # if so set its level appropriately
+                # if not, register it? (maybe)
```

Comment:
Is this a TODO comment?

Reply:
Yeah it's already done, I'll remove

Reply:
Well, I think I'll just not support module names in the set_logs api due to restrictions on kwarg names